standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1187,6 +1187,11 @@ private List<Partition> getPartitionsByQuery(String catName, String dbName, Stri
       MetastoreDirectSqlUtils.timingTrace(doTrace, queryText, start, queryTime);
     }
     String partIds = getIdListForIn(partitions.keySet());
+
+    if (partIds.isEmpty()) {
```

Comment:
do we still need this after you've added iceberg partition list functionality in PartitionUtils?

Reply:
Yes, because the issue with Metastore Direct SQL was encountered after I had added Iceberg partition list functionality in PartitionUtils and also I think PartitionUtils is not used by this metastore code because it is found in hive-ql and it's not a dependency for the metastore modules.

Reply:
👍